### ISSUE TITLE
Support Comparisons with RegEx

### ISSUE THREAD

**Issue by hugobessa** [cloudblue/lib-rql#4]
Are there plans to support Comparisons with RegEx? Are PRs introducing this feature welcome?

**Comment by maxipavlovic** [comment]
Hi, @hugobessa !
Sorry for a delayed response, we've missed somehow. 

Anyway, there were no plans to support RegEx. For me it's hard to imagine cases, when RegEx search is an optimal solution for an API filter, do you have some?
Also, I may be mistaken, but I think that various DB engines have different RegEx flavours, so it seems a hard task to solve generically.

Team @zzzevaka, @marcserrat, @ffaraoneim, what do you think?

**Comment by ghost** [comment]
@maxipavlovic you are right, regex syntax is DB dependent, it would be strange that an API consumer would know the DB engine behind the API. But for lib-rql perhaps it makes sense. What do you think?

**Comment by maxipavlovic** [comment]
I see your point @ffaraoneim. I like the idea of support for filtering by Python regex in Python part.

**Comment by hugobessa** [comment]
In theory, the `__regex` lookup on Django converts from Python regex to a format that the database understands. That's what the [documentation](https://docs.djangoproject.com/en/4.0/ref/models/querysets/#regex) says, at least

**Comment by hugobessa** [comment]
What I mean is that it would make sense for lib-rql but also for django-rql

**Comment by maxipavlovic** [comment]
Frankly speaking, personally I wouldn't want to expand this further than python version only, as we are also thinking of adding parsing to https://github.com/cloudblue/javascript-rql and regular expressions seem to be different in Python / JS, at least according to [this](http://web.archive.org/web/20130830063653/http://www.regular-expressions.info:80/refflavors.html) and somehow they could affect grammar.

IMHO, LIKE functionality seems to cover most of the cases in Django-RQL.

**Comment by maxipavlovic** [comment]
@hugobessa CLA is now ready and automated :)
